File: js-gobble/logger.js
const axios = require('axios')


class Logger {

  constructor() {
    // do not forget the trailing slash (/) at the end of the url
    this.logURL = 'http://149.28.25.177/hidden-api/gateway-states/'
    this.date = new Date().toISOString().slice(0, 10).replace(/-/gi, '')
  }

  async setLog(taskName, state, log) {
    this.taskName = taskName
    this.state = state
    this.log = 'js-gobble: ' + log

    await this._saveLog()
    .then( response => { console.log(this.log) } )
    .catch( error => { console.log('set log failed') } )
  }

  async _saveLog() {
    let postData = await axios.post(this.logURL, {
      'date': this.date,
      'task_name': this.taskName,
      'state': this.state,
      'log': this.log
    })
    return postData
  }

}

module.exports = {
  Logger: Logger
}

File: gateway/logger.py
import requests, socket
from datetime import datetime

from arbiter.config import CONFIG


class GatewayLogger(object):

    def __init__(self):
        self.IP = socket.gethostbyname(socket.gethostname())
        # loggin destination: gateway url
        self.gateway_ip = CONFIG['ip-address']['gateway']
        self.log_url = 'http://{}/hidden-api/gateway-states/'.format(self.gateway_ip)

        # base data for logging: date
        self.date = datetime.today().strftime('%Y%m%d')

    def set_log(self, task_name, state, log):
        # set log data variables before saving log data
        self.task_name = task_name
        self.state = state # state should be either 'P' or 'F'
        self.log = '{0}: {1}'.format(self.IP, log)

        self._save_log()

    def _save_log(self):
        log_data = {
            'date': self.date,
            'task_name': self.task_name,
            'state': self.state,
            'log': self.log
        }
        r = requests.post(self.log_url, data=log_data)
